test/conformance/testing/include/uur/raii.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2022-2023 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef UUR_RAII_H_INCLUDED
+#define UUR_RAII_H_INCLUDED
+
+#include "ur_api.h"
+#include <cstddef>
+
+namespace uur {
+namespace raii {
+template <class URHandleT> inline void release(URHandleT handle);
+#define OVERLOAD(URHandleT, urHandleRelease)                                   \
+    template <> inline void release<URHandleT>(URHandleT handle) {             \
+        urHandleRelease(handle);                                               \
+    }
+OVERLOAD(ur_loader_config_handle_t, urLoaderConfigRelease)
+OVERLOAD(ur_adapter_handle_t, urAdapterRelease)
+OVERLOAD(ur_device_handle_t, urDeviceRelease)
+OVERLOAD(ur_context_handle_t, urContextRelease)
+OVERLOAD(ur_mem_handle_t, urMemRelease)
+OVERLOAD(ur_sampler_handle_t, urSamplerRelease)
+OVERLOAD(ur_usm_pool_handle_t, urUSMPoolRelease)
+OVERLOAD(ur_physical_mem_handle_t, urPhysicalMemRelease)
+OVERLOAD(ur_program_handle_t, urProgramRelease)
+OVERLOAD(ur_kernel_handle_t, urKernelRelease)
+OVERLOAD(ur_queue_handle_t, urQueueRelease)
+OVERLOAD(ur_event_handle_t, urEventRelease)
+#undef OVERLOAD
+
+template <class URHandleT> struct Wrapper {
```

Comment:
You could use a non-type template parameter like I did [here](https://github.com/oneapi-src/unified-runtime/pull/1286/files#diff-d0ae1ee0edc109c505db620df503c365a6daf95b5f4ef9f12ba06a7d16e0b3d8R47).

Reply:
macro deleted.